### PR TITLE
Git submodule scan

### PR DESCRIPTION
**Pull request by mathstuf**
Crude attempt at #147. Does not work. My Ruby-fu is too weak, but should work as a sketch.

### CODE REVIEW

File: ruby/command-t/scanner/file_scanner/git_file_scanner.rb [hunk 1]
```diff
@@ -15,13 +15,40 @@ def paths!
...
+                'git',
+                'submodule',
+                'foreach',
+                'git',
+                'ls-files'
+              ])
+
+              module_name = ''
+              submodule_merger = lambda { |path|
+                if (path =~ /^Entering '(.*)'$/) === 0
```
Comment (mathstuf) [inline ruby/command-t/scanner/file_scanner/git_file_scanner.rb]: Probably needs to be smarter about locale in case the string isn't the same.

**Comment by mathstuf** [comment]
Ping? I'm not really all that familiar with Ruby and have basically hit a roadblock of my knowledge of it. Thoughts about how to do the stream merging of the two?

**Comment by wincent** [comment]
@mathstuf: I'm going to talk a look right now and see if I can beat this into shape. Thanks for the PR!

**Comment by wincent** [comment]
Ok, I think this is now working. I tested it with a project with nested submodules like this:

```
$ git clone some-project
$ cd some-project
$ vim # test Command-T when submodules aren't checked out
$ git submodule update --init
$ vim # test Command-T with top-level submodules checked out
$ git submodule update --init --recursive
$ vim # test Command-T with all submodules checked out
```

It's possible if a submodule is in a bad state that an error cold be thrown in the scanner, which seems slightly more likely than the old version of the scanner throwing an error (because submodules can be in more possible states), but I suspect this is probably ok. We can deal with that if it ever crops up.

I'll let this bake on the `next` branch for a while before merging to master. Thanks once again for submitting the PR.

**Comment by mathstuf** [comment]
Thanks! I'll test it out next week where I use submodules more regularly.

**Pull request closed without merging**